airflow/models/dag.py

```diff
@@ -2828,7 +2842,15 @@ def add_logger_if_needed(ti: TaskInstance):
                 try:
                     add_logger_if_needed(ti)
                     ti.task = tasks[ti.task_id]
-                    _run_task(ti, session=session)
+                    ret = _run_task(ti, session=session)
+                    if ret is TaskReturnCode.DEFERRED:
+                        if not _triggerer_is_healthy():
+                            raise _StopDagTest(
+                                "Task has deferred but triggerer component is not running. "
+                                "You can start the triggerer by running `airflow triggerer` in a terminal."
+                            )
+                except _StopDagTest:
+                    raise
```

Comment:
What does this block do? If this simply reraise there’s no point catching this in the first place.

Reply:
note that the next line catches `Exception` and does not raise.  so if we don't catch and raise here, the exception is suppressed. so we must catch it and reraise in order to not suppress it.

Reply:
Ah I see, you’re right. Maybe a comment here?